visitors/assemble.go

```diff
@@ -0,0 +1,359 @@
+// Copyright 2018 the LinuxBoot Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package visitors
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"log"
+	"sort"
+
+	"github.com/linuxboot/fiano/pkg/lzma"
+	"github.com/linuxboot/fiano/uefi"
+)
+
+// Assemble reconstitutes the firmware tree assuming that the leaf node buffers are accurate
+type Assemble struct {
+}
+
+// Run just applies the visitor.
+func (v *Assemble) Run(f uefi.Firmware) error {
+	return f.Apply(v)
+}
+
+// Visit applies the Assemble visitor to any Firmware type.
+func (v *Assemble) Visit(f uefi.Firmware) error {
+	var err error
+
+	// Get the damn Erase Polarity
+	switch f := f.(type) {
+
+	case *uefi.FirmwareVolume:
+		// Set Erase Polarity
+		uefi.Attributes.ErasePolarity = f.GetErasePolarity()
```

Comment:
Rather than using a global variable, is it possible to store the erase polarity inside of the visitor instead?

Reply:
I could store it in the visitor, but I think I call into code in package uefi that checks that variable so I would have to set it anyway. We should overhaul this, but not in this PR